frontend/src/components/modules/LadderTabs/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { LadderInterface } from './LadderTabs.interface';
+
+export const generateBandGrouping = (
+  totalNumberOfBands: number,
+  bandsPerLevel = 3,
+  positionNames = ['Junior', 'Mid', 'Senior'],
+) => {
+  let bandGrouping: LadderInterface = {};
+
+  let bandCounter = 1; // Initialize band counter
+
+  // Loop through position names
+  for (const position of positionNames) {
+    const bands = [];
+    // Calculate the starting band for the current position
+    const startBand = positionNames.indexOf(position) * bandsPerLevel + 1;
+    // Generate bands for the current position
+    for (let i = 0; i < bandsPerLevel && bandCounter <= totalNumberOfBands; i++) {
+      bands.push(startBand + i);
+      bandCounter++;
+    }
+    // Assign bands to position if bands were generated
+    if (bands.length > 0) {
+      bandGrouping[position] = bands;
+    }
+  }
+
+  return bandGrouping;
```

Comment:
This function seems too complicated for its purpose. I believe it should be simplified

Maybe something like this piece of code? Or maybe you would have an even better idea to make it more readable and simplified 
```suggestion
  export const generateBandGrouping = (
  totalNumberOfBands: number,
  bandsPerLevel = 3,
  positionNames = ['Junior', 'Mid', 'Senior'],
): LadderInterface => {
  const bands = Array.from({ length: totalNumberOfBands }, (_, i) => i + 1);

  return Object.fromEntries(
    positionNames.map((key, index) => {
      const startIndex = index * bandsPerLevel;
      return [key, bands.slice(startIndex, startIndex + bandsPerLevel)];
    }),
  );
};
```

Reply:
I like your idea. Changed